Engine/DataFeeds/SubscriptionFrontierTimeProvider.cs

```diff
@@ -57,6 +57,15 @@ private void UpdateCurrentTime()
             long earlyBirdTicks = MaxDateTimeTicks;
             foreach (var subscription in _subscriptionManager.DataFeedSubscriptions)
             {
+                if (subscription.Current == null
+                    && !subscription.IsUniverseSelectionSubscription
+                    && subscription.UtcStartTime == _utcNow)
+                {
+                    // this is a data subscription we just added
+                    // lets move it next to find the initial emit time
+                    subscription.MoveNext();
```

Comment:
I wonder if this is where this should happen. When we _first_ create the subscription, do we not prime the enumerator stack? I just want to make sure there isn't a chance for performing this initial `MoveNext` twice

Reply:
I can agree with this reasoning. We definitely don't want to force blocking from the algo thread during initialization as it will kill startup times.